packages/desktop/components/TitleBar/index.tsx

```diff
@@ -46,7 +46,8 @@ function TitleBar() {
           className="btn btn-ghost btn-square btn-sm"
           onClick={async () => {
             flushSync(() => {
-              if (window.wowarenalogs.win?.hideToSystemTray) {
+              // the condition below checks whether the app is on a recent enough version that supports system tray
+              if (window.wowarenalogs.win?.hideToSystemTray && window.wowarenalogs.app?.isUpdateAvailable) {
```

Comment:
Why is `isUpdateAvailable` needed here?

Reply:
Because on older versions, the hide-to-system-tray behavior makes it difficult for people to actually properly quit the app, thus they don't really get the app updates

Reply:
kind of like a forcing function to get everyone onto the latest version first.

Reply:
Won't `isUpdateAvailable` be false on a normal boot-up of the most recent app and make `hideToSystemTray()` never get called?

Reply:
this is checking "whether the function is non-null" instead of "calling the function and checking the result" :)